From 9e48501bb1985b541466eceed44e320196efc576 Mon Sep 17 00:00:00 2001
From: Kei Okada <kei.okada@gmail.com>
Date: Fri, 13 Feb 2015 19:54:54 +0900
Subject: [PATCH 1/2] add .catkin to CMakeLists.txt

---
 CMakeLists.txt | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 865ad80..e576565 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,4 +39,30 @@ foreach(target SeqInComp SeqOutComp MyServiceConsumerComp MyServiceProviderComp)
   install(CODE "execute_process(COMMAND cmake -E create_symlink ../../../share/openrtm-1.1/example/${target} ${target} WORKING_DIRECTORY \$ENV{DESTDIR}\${CMAKE_INSTALL_PREFIX}/lib/openrtm_aist/bin)")
 endforeach()
 
+string(REGEX MATCH "catkin" need_catkin "$ENV{_}")
+if(need_catkin)
+  install(CODE "
+## this is tricky force write catkin marker file
+set(_catkin_marker_file \"\${CMAKE_INSTALL_PREFIX}/.catkin\")
+# check if the develspace marker file exists yet
+if(EXISTS \${_catkin_marker_file})
+  file(READ \${_catkin_marker_file} _existing_sourcespaces)
+  if(_existing_sourcespaces STREQUAL \"\")
+    # write this sourcespace to the marker file
+    file(WRITE \${_catkin_marker_file} \"${CMAKE_INSTALL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}\")
+  else()
+    # append to existing list of sourcespaces if it's not in the list
+    list(FIND _existing_sourcespaces \"${CMAKE_INSTALL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}\" _existing_sourcespace_index)
+    if(_existing_sourcespace_index EQUAL -1)
+      file(APPEND \${_catkin_marker_file} \";${CMAKE_INSTALL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}\")
+    endif()
+  endif()
+else()
+  # create a new develspace marker file
+  # NOTE: extra care must be taken when running multiple catkin jobs in parallel
+  #       so that this does not overwrite the result of a similar call in another package
+  file(WRITE \${_catkin_marker_file} \"${CMAKE_INSTALL_PREFIX}/${CATKIN_PACKAGE_SHARE_DESTINATION}\")
+endif()
+")
+endif()
 
-- 
1.9.1

