From 1d628cd2ab387f9aaf658a0b3597b39dc77c95a6 Mon Sep 17 00:00:00 2001
From: Kei Okada <k-okada@jsk.t.u-tokyo.ac.jp>
Date: Tue, 9 Apr 2024 22:50:38 +0900
Subject: [PATCH 7/7] remove duplicate definition of config_fname

---
 src/lib/rtm/Manager.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/lib/rtm/Manager.cpp b/src/lib/rtm/Manager.cpp
index 647bd1f..cc922af 100644
--- a/src/lib/rtm/Manager.cpp
+++ b/src/lib/rtm/Manager.cpp
@@ -1786,7 +1786,6 @@ std::vector<coil::Properties> Manager::getLoadableModules()
     std::string name_conf(category + "." + inst_name + ".config_file");
     coil::vstring config_fname;
     
-    coil::vstring config_fname;
     coil::Properties type_prop, name_prop;
     
     // Load "category.instance_name.config_file"
-- 
2.25.1

